File: src/main/scala/com/vc/easy/L270.scala
package com.vc.easy


class TreeNode270(var _value: Int) {
  var value: Int = _value
  var left: TreeNode270 = null
  var right: TreeNode270 = null
}

object L270 {
  def closestValue(root: TreeNode270, target: Double): Int = {
    var current = root
    var minDistance = Double.MaxValue
    var distance = 0.toDouble
    var res = 0
    while(current != null) {
      distance = Math.abs(current.value.toDouble - target)
      if(distance < minDistance) {
        minDistance = distance
        res = current.value
      }
      if(current.value > target) {
        current = current.left
      }
      else if(current.value < target) {
        current = current.right
      }
      else return current.value
    }
    res
  }
}

File: src/main/java/com/vc/easy/L270.java
package com.vc.easy;

import com.vc.hard.TreeNode;

class L270 {
    public int closestValue(TreeNode root, double target) {
        double minDiff = Double.MAX_VALUE;
        int value = -1;

        TreeNode current = root;
        while(current != null) {
            double diff = Math.abs(target - current.val);
            if(minDiff > diff) {
                minDiff = diff;
                value = current.val;
            }
            if(current.val > target) current = current.left;
            else if(current.val < target) current = current.right;
            else return current.val;
        }
        return value;
    }
}
